framework/assets/yii.validation.js

```diff
@@ -87,48 +87,33 @@ yii.validation = (function ($) {
                     return;
                 }
 
-                var def = $.Deferred(),
-                    fr = new FileReader(),
-                    img = new Image();
-
-                img.onload = function () {
-                    if (options.minWidth && this.width < options.minWidth) {
-                        messages.push(options.underWidth.replace(/\{file\}/g, file.name));
-                    }
-
-                    if (options.maxWidth && this.width > options.maxWidth) {
-                        messages.push(options.overWidth.replace(/\{file\}/g, file.name));
-                    }
-
-                    if (options.minHeight && this.height < options.minHeight) {
-                        messages.push(options.underHeight.replace(/\{file\}/g, file.name));
-                    }
-
-                    if (options.maxHeight && this.height > options.maxHeight) {
-                        messages.push(options.overHeight.replace(/\{file\}/g, file.name));
-                    }
-                    def.resolve();
-                };
-
-                img.onerror = function () {
-                    messages.push(options.notImage.replace(/\{file\}/g, file.name));
-                    def.resolve();
-                };
+                var deferred = $.Deferred();
```

Comment:
Removed reduction. Moved `validateImage` to separate method because it's impossible to test this logic otherwise - local variables exist only in their scope and we can't redefine native event handlers. This for example:

```js
FileReader.prototype.onload = undefined;
```

gives:

```
TypeError: 'set onload' called on an object that does not implement interface FileReader.
```

I tried my best to think a workaround, but seems like moving it to separate method is the best solution to avoid ugly hacks.